tiledb/sm/c_api/tiledb.cc

```diff
@@ -4117,10 +4117,8 @@ int32_t tiledb_array_alloc(
 
   // Allocate an array object
   (*array)->array_ =
-      new (std::nothrow) tiledb::sm::Array(uri, ctx->ctx_->storage_manager());
+      make_shared<tiledb::sm::Array>(HERE(), uri, ctx->ctx_->storage_manager());
   if ((*array)->array_ == nullptr) {
```

Comment:
This check no longer works with the change from `new (std::nothrow)` to `make_shared`. You'll need `try` to detect an error with `make_shared`.